brat/Brat/Checker.hs

```diff
@@ -315,6 +297,24 @@ check' (Th tm) ((), u@(hungry, ty):unders) = case (?my, ty) of
       _ -> err . ExpectedThunk "" $ showRow (u:unders)
     pure (((), ()), ((), unders))
   (Kerny, _) -> err . ThunkInKernel $ show (Th tm)
+ where
+  checkThunk :: forall m. (CheckConstraints m UVerb, EvMode m)
```

Comment:
you should be be able to get rid of the `forall m` and `Modey m` arguments. The right `m` is bound by `check` and will be used here by default thanks to the `ScopedTypeVariables` extension

Reply:
Just checked again and I'm way wrong about this! Merge away